### DISCUSSION TITLE
Hopper connected to network does not exist

### DISCUSSION THREAD

**Discussion by Nulled-Pointer** [cc-tweaked/CC-Tweaked#2030] [Q&A] [answered]
I have 2 hoppers connected to my computer, and both can be wrapped correctly by `peripheral.wrap()` but neither of them can push or pull to each other, giving an error like this
![image](https://github.com/user-attachments/assets/69d218a2-66be-425c-92d4-d3bdde39ad05)

This is my code at the moment:
```lua
local input = peripheral.wrap("minecraft:hopper_4")
local output = peripheral.wrap("minecraft:hopper_3")

while true do
    output.pushItems(peripheral.getName(input),1)
    sleep(0.5)
```
I also tried doing the reverse (pulling from the output) and directly giving the strings for the hoppers to the function, but they all give similar errors

I did correctly connect the hoppers to the computer to my knowledge
![image](https://github.com/user-attachments/assets/08df3c40-9808-43aa-bf59-e2647f8ce9c4)

**Comment by SquidDev** [comment, marked as answer]
Ahh, this appears to be another instance of #1599. Due to how CC:T's modems are implemented, the two hoppers are on different networks and so can't see each other. If you move the computer to only be next to one of the modems, and then connect the two modems with a cable, this should work.

**Comment by Nulled-Pointer** [comment]
that worked, thanks
